### ISSUE TITLE
missing-input-response and missing-input-secret

### ISSUE THREAD

**Issue by kevinorriss** [ixudra/curl#17]
```
        $captcha = Curl::to('https://www.google.com/recaptcha/api/siteverify')
            ->withData(array(
                'secret' => 'my_secret', 
                'response' => $recaptcha_response,
                'remoteip' => $_SERVER['REMOTE_ADDR']
            ))->asJson(true)->post();
```
I get...

```
[error-codes] => Array
        (
            [0] => missing-input-response
            [1] => missing-input-secret
        )
```

I can get this working using curl-less code... so I am not sure why this seems to not send my post data?

**Comment by elimentz** [comment]
As far as I can see, there is nothing wrong with your request, syntax wise anyways. It's possible that you still need to add a couple of other cURL options, but I haven't used this API so that's just a guess on my part.

When you say 'curl-less code`, you mean simply using curl and not the package? If so, can you show me an example of how you did manage to get it to work?

**Comment by kevinorriss** [comment]
```
$url = 'https://www.google.com/recaptcha/api/siteverify';
        $data = array(
            'secret' => 'my_secret', 
            'response' => $recaptcha_response,
            'remoteip' => $_SERVER['REMOTE_ADDR']
        );
        $options = array(
            'http' => array(
                'header'  => "Content-type: application/x-www-form-urlencoded\r\n",
                'method'  => 'POST',
                'content' => http_build_query($data)
            )
        );
        $context  = stream_context_create($options);
        $result = file_get_contents($url, false, $context);
        if ($result === FALSE) {
            // handle error
        }
        $captcha = json_decode($result, TRUE);
```

**Comment by elimentz** [comment]
Could this be a solution for you? http://stackoverflow.com/questions/29905806/google-recaptcha-2-0-not-working-is-it-for-curl

**Comment by kevinorriss** [comment]
I don't see how, the code I posted works fine and doesn't need curl extension enabled. But it would be nice if the Curl project worked for me as the code call is cleaner. I even added the cacert.pem file to my php.ini file but no luck. Might output the details to a log file in the morning to see what is going on.

**Comment by elimentz** [comment]
Let me know how it goes. From what I can tell, there is no reason why the package shouldn't work so the only thing I can think of is something related to specific cURL options or something along those lines...

**Comment by kevinorriss** [comment]
Will do, yeah I didn't see any other options in the ini file for curl. I might take a look at creating my own API the same as this but not using curl.

**Comment by kevinorriss** [comment]
I found the problem... the asJson() function does too much. If used, the response is not only json decoded, but the request is json encoded. Google's reCAPTCHA does not accept the request in json format, but it does return a json string. The code needs something like this

```
->asJsonRequest()
->asJsonResponse(true) -- where boolean is asAssociativeArray
```

to separate the two options.

For now I am simply removing asJson and doing it manually afterwards and it works.

Hope this helps :)

**Comment by elimentz** [comment]
Thanks for letting me know, I will fix this asap

**Comment by kevinorriss** [comment]
You're welcome, I look forward to using the updated version

**Comment by elimentz** [comment]
Fixed in version 6.3.0:

```php
        $response = \Curl::to('http://someurl.com')
            ->withData( $data )
            ->asJsonRequest()
            ->asJsonResponse()
            ->get();
```

The `asJson()` will keep functioning as it has been doing. It will be a shortcut for using both the `asJsonRequest` and the `asJsonResponse` method behind the scenes
